handlers.go

```diff
@@ -0,0 +1,247 @@
+	AddVsocksHandler,
+)
+
+var defaultHandlers = Handlers{
+	Validation: defaultValidationHandlerList,
+	FcInit:     defaultFcInitHandlerList,
+}
+
+// Handlers is a container that houses categories of handler lists.
+type Handlers struct {
+	Validation HandlerList
+	FcInit     HandlerList
+}
+
+// Run will execute all handlers in the Handlers object by flattening the lists
+// into a single list and running.
+func (h Handlers) Run(ctx context.Context, m *Machine) error {
+	l := HandlerList{}.Append(
+		h.Validation.list...,
+	).Append(
+		h.FcInit.list...,
+	)
+
+	return l.Run(ctx, m)
+}
+
+// HandlerList represents a list of named handler that can be used to execute a
+// flow of instructions for a given machine.
+type HandlerList struct {
+	list []Handler
+}
+
+// Append will append a new handler to the handler list.
+func (l HandlerList) Append(handlers ...Handler) HandlerList {
+	l.list = append(l.list, handlers...)
+
+	return l
+}
+
+// Len return the length of the given handler list
+func (l HandlerList) Len() int {
+	return len(l.list)
+}
+
+// Has will iterate through the handler list and check to see if the the named
+// handler exists.
+func (l HandlerList) Has(name string) bool {
+	for _, h := range l.list {
+		if h.Name == name {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Swap will replace all elements of the given name with the new handler.
+func (l HandlerList) Swap(handler Handler) HandlerList {
+	newList := HandlerList{}
+	for _, h := range l.list {
+		if h.Name == handler.Name {
+			newList.list = append(newList.list, handler)
+			continue
+		}
+
+		newList.list = append(newList.list, h)
+	}
+
+	return newList
+}
+
+// Swappend will either append, if there isn't an element within the handler
+// list, otherwise it will replace all elements with the given name.
+func (l HandlerList) Swappend(handler Handler) HandlerList {
+	if l.Has(handler.Name) {
+		return l.Swap(handler)
+	}
+
+	return l.Append(handler)
+}
+
+// Remove will return an updated handler with all instances of the specific
+// named handler being removed.
+func (l HandlerList) Remove(name string) HandlerList {
+	newList := HandlerList{}
+	for _, h := range l.list {
+		if h.Name != name {
+			newList.list = append(newList.list, h)
+		}
+	}
+
+	return newList
+}
+
+// Clear clears all named handler in the list.
+func (l HandlerList) Clear() HandlerList {
+	l.list = l.list[0:0]
+	return l
+}
+
+// Handler represents a named handler that contains a name and a function which
+// is used to execute during the initialization process of a machine.
+type Handler struct {
+	Name string
+	Fn   func(context.Context, *Machine) error
+}
+
+// Run will execute each instruction in the handler list. If an error occurs in
+// any of the handlers, then the list will halt execution and return the error.
+func (l HandlerList) Run(ctx context.Context, m *Machine) error {
+	for _, handler := range l.list {
+		if err := handler.Fn(ctx, m); err != nil {
```

Comment:
If these are always going to be run this way would it make sense to add the log message that says "executing handler x" here instead of inside the handlers for consistency if someone adds custom handlers?

Reply:
Ah, I think that got squashed when rebasing. Thanks! Will refix.